pkg/transparentproxy/validate/validator_test.go

```diff
@@ -0,0 +1,74 @@
+package validate
+
+import (
+	"net/netip"
+	"os"
+	"strings"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	"github.com/kumahq/kuma/pkg/core"
+	kuma_log "github.com/kumahq/kuma/pkg/log"
+)
+
+var _ = Describe("Should Validate iptables rules", func() {
+	Describe("generate default validator config", func() {
+		It("ipv4", func() {
+			// when
+			validator := createValidator(false)
+
+			// then
+			Expect(validator.Config.ServerListenIP.String()).To(Equal("127.0.0.1"))
+			Expect(validator.Config.ServerListenPort).To(Equal(uint16(15006)))
+		})
+
+		It("ipv6", func() {
+			// when
+			validator := createValidator(true)
+
+			// then
+			serverIP := validator.Config.ServerListenIP.String()
+			Expect(serverIP).To(Equal("::1"))
+
+			splitByCon := strings.Split(serverIP, ":")
+			Expect(len(splitByCon)).To(BeNumerically(">", 2))
+		})
+	})
+
+	It("should return pass when connect to correct address", func() {
+		// when
+		validator := createValidator(false)
+		ipAddr := "127.0.0.1"
+		addr, _ := netip.ParseAddr(ipAddr)
+		validator.Config.ServerListenIP = addr
+		validator.Config.ClientConnectIP = addr
+		validator.Config.ClientConnectPort = ValidationServerPort
+
+		err := validator.Run()
+
+		// then
+		Expect(err).ToNot(HaveOccurred())
+	})
+
+	It("should return fail when no iptables rules setup", func() {
+		// given
+		validator := createValidator(false)
+		validator.Config.ClientRetryInterval = 30 * time.Millisecond
```

Comment:
Not sure I understand this test completely. There is no iptables injection going on in the test we just make te retry interval so small that it fails, right? Shouldn't this be even lower to be less sensitive to flakes?

Reply:
Right, we are not injecting any iptables rules in the test cases, the cases here are only verifying the logic of the validator itself should behave correctly 
I think 30ms is a reasonable value for testing just to make it faster and there should be no flakiness here because the connection will never establish successfully without the redirection set up by the iptables rules.

Reply:
we could add a comment in the code that "the connection will never establish successfully" and that this value does not matter that much as long as it's low so the test is fast, right?

Reply:
```suggestion
		validator.Config.ClientRetryInterval = 30 * time.Millisecond // just to make test faster and there should be no flakiness here because the connection will never establish successfully without the redirection 
```